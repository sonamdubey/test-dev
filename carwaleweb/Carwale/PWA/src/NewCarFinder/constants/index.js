/**
 * API URL's
 */
export const SUGGESTED_BUDGET_ENDPOINT = "/api/newcarfinder/budget/"
export const NEWCAR_SEARCH_ENDPOINT = "/api/newcarfinder/search/"
export const SUGGESTED_BODYTYPE_ENDPOINT = "/api/newcarfinder/bodytype/"
export const FUELTYPE_ENDPOINT = "/api/newcarfinder/fueltype/"
export const MAKE_ENDPOINT = "/api/newcarfinder/filters/make/"
export const VERSION_CITY_PRICE_ENDPOINT = "/api/prices/version/"
export const DEFAULT_FILTERS_ENDPOINT = "/api/filters/all/"
export const FILTER_SCREEN_ENDPOINT = "/api/newcarfinder/screen/"

/**
 * PAGE URL's
 */
export const NEWCARFINDER_ENDPOINT = '/find-car/'
export const NEWCARFINDER_FILTERS_ENDPOINT = '/find-car/filters/'
export const NEWCARFINDER_RESULTS_ENDPOINT = '/find-car/results/'
export const NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"body-type/"
export const NEWCARFINDER_FILTERS_FUELTYPE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"fuel-type/"
export const NEWCARFINDER_FILTERS_MAKE_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"make/"
export const NEWCARFINDER_FILTERS_BUDGET_ENDPOINT = NEWCARFINDER_FILTERS_ENDPOINT+"budget/"
/**
 * SEO Pages URL's
 */
export const NEWCARFINDER_SEO_SINGLE_DIMENSION_ENDPOINT = '/best-([a-zA-Z]+)-cars-in-india/'

/**
 * NCF CONSTANTS
*/
export const BUDGET_SLIDER_MIN = 200000
export const BUDGET_SLIDER_MAX = 10000000
export const CATEGORY_NAME = "NCFPWA"
export const PAGE_VIEW_CATEGORY = "PageViews"
export const SEO_TYPE = {
    SEO_SINGLE: 'SEO_SINGLE'
}
