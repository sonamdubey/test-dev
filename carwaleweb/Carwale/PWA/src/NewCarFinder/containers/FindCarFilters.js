import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import FindCarFilterRouter from '../routes/FindCarFilters'
import FiltersFooter from './FiltersFooter'
import {
	hideFooter
} from '../../actionCreators/Footer'
import {
    hideHeader
} from '../../actionCreators/Header'
import {
    hideShortlistIcon,makeHeaderNormal
} from '../actionCreators/HeaderWrapper'

class FindCarFilters extends React.Component {
	constructor(props) {
		super(props)
	}

	componentWillMount(){
		this.props.hideShortlistIcon()
		this.props.makeHeaderNormal()
	}
	render() {
			return (
			<div>
				<FindCarFilterRouter />
				<FiltersFooter />
			</div>
			)
		}
}

const mapDispatchToProps = (dispatch) => {
    return {
        hideFooter: bindActionCreators(hideFooter, dispatch),
		hideHeader: bindActionCreators(hideHeader, dispatch),
		hideShortlistIcon: bindActionCreators(hideShortlistIcon, dispatch),
		makeHeaderNormal: bindActionCreators(makeHeaderNormal, dispatch)
    }
}


export default connect(null,mapDispatchToProps)(FindCarFilters)
