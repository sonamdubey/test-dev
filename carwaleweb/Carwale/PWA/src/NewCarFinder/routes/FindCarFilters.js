import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Make from '../containers/Make'
import BodyType from '../containers/BodyType'
import FuelType from '../containers/FuelType'
import Budget from '../containers/Budget'
import {
	NEWCARFINDER_FILTERS_MAKE_ENDPOINT,
	NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT,
	NEWCARFINDER_FILTERS_FUELTYPE_ENDPOINT,
	NEWCARFINDER_FILTERS_BUDGET_ENDPOINT,
} from '../constants/index'

class FindCarFilterRouter extends React.Component {
	render() {
		return (
			<Switch>
				<Route path={NEWCARFINDER_FILTERS_MAKE_ENDPOINT} component={Make} />
				<Route path={NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT} component={BodyType} />
				<Route path={NEWCARFINDER_FILTERS_FUELTYPE_ENDPOINT} component={FuelType} />
				<Route exact path={NEWCARFINDER_FILTERS_BUDGET_ENDPOINT} component = { Budget } />
			</Switch>
		)
	}
}

export default FindCarFilterRouter
