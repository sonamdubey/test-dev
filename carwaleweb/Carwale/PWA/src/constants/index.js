/**
 * API URL's
 */
export const CITY_AUTOCOMPLETE_URL = "/api/v2/autocomplete/city/"
export const AREA_AUTOCOMPLETE_URL = "/api/v2/autocomplete/areas/"


/**
 * UNIQUE ID's
 */
export const UNIVERSAL_ANALYTICS_ID = "UA-337359-1"


/**
 * App download links
 */
export const APP_STORE_LINK = 'https://itunes.apple.com/in/app/carwale/id910137745?mt=8'
export const PLAY_STORE_FOOTER_LINK = 'https://play.google.com/store/apps/details?id=com.carwale&referrer=utm_source%3DCarWaleMsite%26utm_medium%3DFooter%26utm_campaign=CarWale%2520MobilesiteFooter'
export const PLAY_STORE_NAV_DRAWER_LINK = 'https://play.google.com/store/apps/details?id=com.carwale&referrer=utm_source%3DCarWaleMsite%26utm_medium%3DMenuDrawer%26utm_campaign=CarWale%2520MobilesiteMenuDrawer'


export const desktopBreakPoint = 780
