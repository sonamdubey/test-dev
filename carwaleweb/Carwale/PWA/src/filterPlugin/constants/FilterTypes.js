const filterTypes = {
    budget : 1,
    fuelType : 2,
    bodyType : 3,
    make : 4,
    transmissionType : 5
}

export default filterTypes
