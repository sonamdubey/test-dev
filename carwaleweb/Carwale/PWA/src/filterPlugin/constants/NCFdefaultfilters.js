import snapPoints from '../utils/rheostat/constants/budgetSnapPoints'

const ncfDefaultFilters = () => {
    return {
        location: {
            cityId: -1,
            cityName: "",
            userConfirmed: false
        },
        newCarFinder: {
            budget: {
                displayName: "Budget",
                slider: {
                    min: 200000,
                    max: 10000000,
                    values: [
                        0
                    ],
                    userChange: false,
                    snapPoints
                },
                inputBox: {
                    value: 0
                }
            },
            bodyType: {
                displayName: "Body Type",
                data: [
                    {
                        id: 6,
                        name: 'SUV/MUV',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/suv_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/suv.svg',
                        isSelected: false
                    },
                    {
                        id: 3,
                        name: 'Hatchback',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/hatchback_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/hatchback.svg',
                        isSelected: false
                    },
                    {
                        id: 1,
                        name: 'Sedan',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/sedan_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/sedan.svg',
                        isSelected: false
                    },
                    {
                        id: 5,
                        name: 'Convertible',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/convertible_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/convertible.svg',
                        isSelected: false
                    },
                    {
                        id: 2,
                        name: 'Coupe',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/coupe_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/coupe.svg',
                        isSelected: false
                    },
                    {
                        id: 10,
                        name: 'Compact Sedan',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/compactsedan_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/compactsedan.svg',
                        isSelected: false
                    },
                    {
                        id: 4,
                        name: 'Minivan/Van',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/van_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/van.svg',
                        isSelected: false
                    },
                    {
                        id: 8,
                        name: 'Station Wagon',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/wagon_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/wagon.svg',
                        isSelected: false
                    },
                    {
                        id: 7,
                        name: 'Truck',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/truck_clr.svg',
                        lineIcon: 'https://imgd.aeplcdn.com/0x0/cw/body/svg/truck.svg',
                        isSelected: false
                    }
                ]
            },
            fuelType: {
                displayName: "Fuel Type",
                data: [
                    {
                        id: 1,
                        name: 'Petrol',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/petrol.svg',
                        isSelected: false
                    },
                    {
                        id: 2,
                        name: 'Diesel',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/diesel.svg',
                        isSelected: false
                    },
                    {
                        id: 3,
                        name: 'CNG',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/cng.svg',
                        isSelected: false
                    },
                    {
                        id: 5,
                        name: 'Electric',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/fuel/svg/electric.svg',
                        isSelected: false
                    }
                ]
            },
            make: {
                displayName: "Make",
                data: [
                    {
                        "makeId": 10,
                        "makeName": "Maruti Suzuki",
                        "isSelected": false
                    },
                    {
                        "makeId": 8,
                        "makeName": "Hyundai",
                        "isSelected": false
                    },
                    {
                        "makeId": 7,
                        "makeName": "Honda",
                        "isSelected": false
                    },
                    {
                        "makeId": 17,
                        "makeName": "Toyota",
                        "isSelected": false
                    },
                    {
                        "makeId": 16,
                        "makeName": "Tata",
                        "isSelected": false
                    },
                    {
                        "makeId": 9,
                        "makeName": "Mahindra",
                        "isSelected": false
                    },
                    {
                        "makeId": 5,
                        "makeName": "Ford",
                        "isSelected": false
                    },
                    {
                        "makeId": 1,
                        "makeName": "BMW",
                        "isSelected": false
                    },
                    {
                        "makeId": 15,
                        "makeName": "Skoda",
                        "isSelected": false
                    },
                    {
                        "makeId": 11,
                        "makeName": "Mercedes-Benz",
                        "isSelected": false
                    },
                    {
                        "makeId": 20,
                        "makeName": "Volkswagen",
                        "isSelected": false
                    },
                    {
                        "makeId": 18,
                        "makeName": "Audi",
                        "isSelected": false
                    },
                    {
                        "makeId": 4,
                        "makeName": "Fiat",
                        "isSelected": false
                    },
                    {
                        "makeId": 12,
                        "makeName": "Mitsubishi",
                        "isSelected": false
                    },
                    {
                        "makeId": 45,
                        "makeName": "Renault",
                        "isSelected": false
                    },
                    {
                        "makeId": 21,
                        "makeName": "Nissan",
                        "isSelected": false
                    },
                    {
                        "makeId": 23,
                        "makeName": "Land Rover",
                        "isSelected": false
                    },
                    {
                        "makeId": 44,
                        "makeName": "Jaguar",
                        "isSelected": false
                    },
                    {
                        "makeId": 19,
                        "makeName": "Porsche",
                        "isSelected": false
                    },
                    {
                        "makeId": 54,
                        "makeName": "Ssangyong",
                        "isSelected": false
                    },
                    {
                        "makeId": 37,
                        "makeName": "Volvo",
                        "isSelected": false
                    },
                    {
                        "makeId": 51,
                        "makeName": "Mini",
                        "isSelected": false
                    },
                    {
                        "makeId": 22,
                        "makeName": "Bentley",
                        "isSelected": false
                    },
                    {
                        "makeId": 34,
                        "makeName": "Lexus",
                        "isSelected": false
                    },
                    {
                        "makeId": 50,
                        "makeName": "Force Motors",
                        "isSelected": false
                    },
                    {
                        "makeId": 33,
                        "makeName": "Ferrari",
                        "isSelected": false
                    },
                    {
                        "makeId": 43,
                        "makeName": "Jeep",
                        "isSelected": false
                    },
                    {
                        "makeId": 49,
                        "makeName": "Aston Martin",
                        "isSelected": false
                    },
                    {
                        "makeId": 47,
                        "makeName": "Bugatti",
                        "isSelected": false
                    },
                    {
                        "makeId": 56,
                        "makeName": "Datsun",
                        "isSelected": false
                    },
                    {
                        "makeId": 61,
                        "makeName": "Isuzu",
                        "isSelected": false
                    },
                    {
                        "makeId": 30,
                        "makeName": "Lamborghini",
                        "isSelected": false
                    },
                    {
                        "makeId": 36,
                        "makeName": "Maserati",
                        "isSelected": false
                    },
                    {
                        "makeId": 25,
                        "makeName": "Rolls-Royce",
                        "isSelected": false
                    },
                    {
                        "makeId": 66,
                        "makeName": "DC",
                        "isSelected": false
                    },
                    {
                        "makeId": 70,
                        "makeName": "Kia",
                        "isSelected": false
                    }
                ]
            },
            transmissionType: {
                displayName: "Transmission Type",
                data: [
                    {
                        id: 1,
                        name: 'Automatic',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/transmission/svg/automatic_clr.svg?v=1',
                        isSelected: false
                    },
                    {
                        id: 2,
                        name: 'Manual',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/transmission/svg/manual_clr.svg?v=1',
                        isSelected: false
                    },
                    {
                        id: 3,
                        name: 'Semi - Automatic',
                        icon: 'https://imgd.aeplcdn.com/0x0/cw/transmission/svg/semiautomatic_clr.svg?v=1',
                        isSelected: false
                    }
                ]
            },
            seats: {
                displayName: "Seating Capacity",
                data: [
                    {
                        id: 5,
                        name: '5',
                        isSelected: false
                    },
                    {
                        id: 6,
                        name: '6',
                        isSelected: false
                    },
                    {
                        id: 7,
                        name: '7',
                        isSelected: false
                    },
                    {
                        id: 8,
                        name: '8',
                        isSelected: false
                    },
                    {
                        id: 9,
                        name: '8+',
                        isSelected: false
                    }
                ]
            }
        }
    }
}

export default ncfDefaultFilters
