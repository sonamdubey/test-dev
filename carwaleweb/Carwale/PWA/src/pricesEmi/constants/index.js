
/**
 * PAGE URL's
 */

import {
	isDesktop
} from '../../utils/Common';

export const EMIPRICES_ENDPOINT = '/emi-prices/';
export const trackingCategory = isDesktop ? "Desktop_EMICalculatorPage" : "MSite_EMICalculatorPage";
