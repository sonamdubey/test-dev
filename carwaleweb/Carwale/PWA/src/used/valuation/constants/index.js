/**
 * PAGE URL's
 */
export const VALUATION_ENDPOINT = '/used/carvaluation/';

export const BUY_CAR_ID = 1
export const SELL_CAR_ID = 2

// kilometer
export const MIN_KILOMETER_DRIVEN = 1
export const MAX_KILOMETER_DRIVEN = 999999

// api end points
export const MAKE_END_POINT='/webapi/carmakesdata/getcarmakes/'
export const MODEL_END_POINT='/webapi/carmodeldata/GetCarModelsByType/'
export const VERSION_END_POINT='/webapi/carversionsdata/GetCarVersions/'
export const VALUATION_HTML_END_POINT='/m/used/valuation/v1/report/'
export const CITY_MASKING_NAME_END_POINT='/api/city/'
export const VERSION_DETAILS_END_POINT='/webapi/CarVersionsData/GetCarDetailsByVersionId/'
export const MODEL_DETAILS_END_POINT = '/webapi/CarModelData/GetCarDetailsByModelId/'

//manufacturing year
export const DEFAULT_MANUFACTURING_YEAR = 1997
export const MIN_MANUFACTURING_YEAR = 1998
export const MAX_MANUFACTURING_YEAR = new Date().getFullYear()

//Report CTA button text
export const BUY_CAR_BUTTON_TEXT = 'View Similar Cars'
export const SELL_CAR_BUTTON_TEXT = 'Sell Your Car'
