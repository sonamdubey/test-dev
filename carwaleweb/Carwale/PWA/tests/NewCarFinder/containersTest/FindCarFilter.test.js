import React from 'react'
import { Redirect, MemoryRouter } from 'react-router-dom'

import FindCarFilters from '../../../src/NewCarFinder/containers/FindCarFilters'
import { NEWCARFINDER_ENDPOINT, NEWCARFINDER_FILTERS_ENDPOINT, NEWCARFINDER_FILTERS_BUDGET_ENDPOINT, NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT } from '../../../src/NewCarFinder/constants/index';


describe("Tests for FindCarFilters container component",()=>{

    let state
    let props
    let store
    let enzymeWrapper

    const Wrapper = () => {
        if(!enzymeWrapper){
            store = mockStore(state)
            //https://stackoverflow.com/questions/44204828/testing-react-component-enclosed-in-withrouter-preferably-using-jest-enzyme
            //https://github.com/ReactTraining/react-router/blob/v4.1.1/packages/react-router/modules/withRouter.js
            //https://github.com/reactjs/react-redux/blob/master/docs/api.md#static-properties
            //https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over
            /**
             * withRouter has static property WrappedComponent which refers to the component passed to it.
             * But since withRouter uses hoistStatics to copy statics over from passed component and redux connected component also defines
             * a static WrappedComponent refering to the component passed to connect, the WrappedComponent property of connected component is
             * copied over to the WrappedComponent property of the withRouter component
             * e.g. const enhancedComponent = withRouter(connect(mapStateToProps)(DummyComponent))
             *      enhancedComponent.WrappedComponent === DummyComponent //true
             */
            enzymeWrapper = shallowWithStore(<FindCarFilters.WrappedComponent {...props}/>, store)

        }
        return enzymeWrapper
    }

    beforeEach( () => {
        props = undefined
        state = undefined
        enzymeWrapper = undefined
        store = undefined
    })
    describe("when user location is not confirmed",() => {
        it("should render Redirect", () => {

            props = {
                history: {
                    replace:jest.fn()
                },
                location:{
                    pathname: NEWCARFINDER_FILTERS_ENDPOINT
                },
                locationConfirmed: false,
                // since we will be mounting only the innermost component, these have to be mocked
                hideHeader : jest.fn(),
                hideFooter : jest.fn()
            }
            const wrapper = Wrapper()

            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when user location is  confirmed",() => {
        it("should render FindCarFilterRouter and FiltersFooter", () => {

            props = {
                history: {
                    replace:jest.fn()
                },
                location:{
                    pathname: NEWCARFINDER_FILTERS_ENDPOINT
                },
                locationConfirmed: true,
                hideHeader : jest.fn(),
                hideFooter : jest.fn()
            }
            const wrapper = Wrapper()

            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("ShouldComponentUpdate",() => {
        beforeEach(() => {
            props = {
                location:{
                    pathname: NEWCARFINDER_FILTERS_BUDGET_ENDPOINT
                },
                locationConfirmed: true,
                hideHeader : jest.fn(),
                hideFooter : jest.fn()
            }
        })
        it("should return true if path changes", () => {
            const wrapper = Wrapper()
            const nextProps = {
                ...props,
                location : {
                    pathname: NEWCARFINDER_FILTERS_BODYTYPE_ENDPOINT
                }
            }
            expect(wrapper.instance().shouldComponentUpdate(nextProps)).toBeTruthy()
        })
    })
})
