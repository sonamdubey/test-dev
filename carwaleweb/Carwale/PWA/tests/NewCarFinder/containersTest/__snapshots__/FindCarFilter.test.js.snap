// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for FindCarFilters container component when user location is  confirmed should render FindCarFilterRouter and FiltersFooter 1`] = `
<div>
  <FindCarFilterRouter />
  <withRouter(Connect(FiltersFooter)) />
</div>
`;

exports[`Tests for FindCarFilters container component when user location is not confirmed should render Redirect 1`] = `
<Redirect
  push={false}
  to="/find-car/"
/>
`;
